tests/integ/sagemaker/workflow/test_model_steps.py

```diff
@@ -0,0 +1,1138 @@
+        instance_type=instance_type,
+        sagemaker_session=pipeline_session,
+    )
+    train_step_args = pytorch_estimator.fit(inputs=inputs)
+    step_train = TrainingStep(
+        name="pytorch-train",
+        step_args=train_step_args,
+    )
+    model = Model(
+        image_uri=pytorch_estimator.training_image_uri(),
+        model_data=step_train.properties.ModelArtifacts.S3ModelArtifacts,
+        sagemaker_session=pipeline_session,
+        role=role,
+    )
+    regis_model_step_args = model.register(
+        content_types=["*"],
+        response_types=["application/json"],
+        inference_instances=["ml.m5.xlarge"],
+        transform_instances=["ml.m5.xlarge"],
+        description="test-description",
+    )
+    step_model_regis = ModelStep(
+        name="pytorch-register-model",
+        step_args=regis_model_step_args,
+    )
+    create_model_step_args = model.create(
+        instance_type="ml.m5.large",
+        accelerator_type="ml.eia1.medium",
+    )
+
+    step_model_create = ModelStep(
+        name="pytorch-model",
+        step_args=create_model_step_args,
+    )
+
+    step_cond = ConditionStep(
+        name="cond-good-enough",
+        conditions=[
+            ConditionGreaterThanOrEqualTo(left=good_enough_input, right=1),
+        ],
+        if_steps=[step_model_regis],
+        else_steps=[step_model_create],
+        depends_on=[step_train],
+    )
+
+    pipeline = Pipeline(
+        name=pipeline_name,
+        parameters=[
+            good_enough_input,
+            instance_count,
+            instance_type,
+        ],
+        steps=[step_train, step_cond],
+        sagemaker_session=pipeline_session,
+    )
+    try:
+        pipeline.create(role)
+        execution1 = pipeline.start(parameters={})
+        execution2 = pipeline.start(parameters={"GoodEnoughInput": 0})
+        try:
+            execution1.wait(delay=30, max_attempts=60)
+        except WaiterError:
+            pass
+        execution1_steps = execution1.list_steps()
+        for step in execution1_steps:
+            assert not step.get("FailureReason", None)
+            assert step["StepStatus"] == "Succeeded"
+            if _REGISTER_MODEL_NAME_BASE in step["StepName"]:
+                assert step["Metadata"][_REGISTER_MODEL_TYPE]
+        assert len(execution1_steps) == 3
+
+        try:
+            execution2.wait(delay=30, max_attempts=60)
+        except WaiterError:
+            pass
+        execution2_steps = execution2.list_steps()
+        for step in execution2_steps:
+            assert not step.get("FailureReason", None)
+            assert step["StepStatus"] == "Succeeded"
+            if _CREATE_MODEL_NAME_BASE in step["StepName"]:
+                assert step["Metadata"][_CREATE_MODEL_TYPE]
+        assert len(execution2_steps) == 3
+    finally:
+        try:
+            pipeline.delete()
+        except Exception:
+            pass
+
+
+def test_mxnet_model_registration(
```

Comment:
`test_mxnet_model_registration_with_custom_inference`

No runtime repack should happen here.

Reply:
Updated the name and added a comment to explain in the new commit.